vendor/parser-extended/parsetree.mli

```diff
@@ -488,38 +488,29 @@ and binding_op =
     pbop_loc : Location.t;
   }
 
-and function_param_desc =
-  | Pparam_val of arg_label * expression option * pattern
-  (** [Pparam_val (lbl, exp0, P)] represents the parameter:
-      - [P]
-        when [lbl] is {{!Asttypes.arg_label.Nolabel}[Nolabel]}
-        and [exp0] is [None]
-      - [~l:P]
-        when [lbl] is {{!Asttypes.arg_label.Labelled}[Labelled l]}
-        and [exp0] is [None]
-      - [?l:P]
-        when [lbl] is {{!Asttypes.arg_label.Optional}[Optional l]}
-        and [exp0] is [None]
-      - [?l:(P = E0)]
-        when [lbl] is {{!Asttypes.arg_label.Optional}[Optional l]}
-        and [exp0] is [Some E0]
-
-      Note: If [E0] is provided, only
-      {{!Asttypes.arg_label.Optional}[Optional]} is allowed.
-  *)
-  | Pparam_newtype of string loc list
-  (** [Pparam_newtype x] represents the parameter [(type x y z)].
-      [x] carries the location of the identifier, whereas the [pparam_loc]
-      on the enclosing [function_param] node is the location of the [(type x y z)]
-      as a whole.
+and param_val = arg_label * expression option * pattern
+  (** - [P] when [lbl] is [Nolabel] and [exp0] is [None]
+      - [~l:P] when [lbl] is [Labelled l] and [exp0] is [None]
+      - [?l:P] when [lbl] is [Optional l] and [exp0] is [None]
+      - [?l:(P = E0)] when [lbl] is [Optional l] and [exp0] is [Some E0]
+
+      Note: If [E0] is provided, only [Optional] is allowed.
   *)
 
-and function_param =
+and param_newtype = string loc list
+  (** [(type x y z)]. *)
+
+and 'a function_param =
   {
     pparam_loc : Location.t;
-    pparam_desc : function_param_desc;
+    pparam_desc : 'a;
   }
 
+and expr_function_param =
+  [ `Param_val of param_val | `Param_newtype of param_newtype ] function_param
+
+and class_function_param = [ `Param_val of param_val ] function_param
```

Comment:
To reduce the diff a little:

```suggestion
and expr_function_param =
  param_val_or_newtype function_param

and class_function_param = param_val function_param
```

Reply:
I modified the types in 69f1210, let me know what you think

Reply:
The new types look good!